project_material/models/project_task_material.py

```diff
@@ -103,6 +105,9 @@ def unlink(self):
             line.sudo()._cancel_procurements()
         return super().unlink()
 
+    def _should_generate_procurement(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-project-addons/pullRequest?prid=4801852)